src/sentry/static/sentry/app/views/settings/incidentRules/ruleForm/index.tsx

```diff
@@ -274,13 +277,27 @@ class RuleFormContainer extends AsyncComponent<Props, State> {
   handleFieldChange = (name: string, value: unknown) => {
     if (['timeWindow', 'aggregation'].includes(name)) {
       this.setState({[name]: value});
+      if (name === 'aggregation') {
+        if (value === 0) {
+          this.props.rule.aggregate = 'count()';
+        } else if (value === 1) {
+          this.props.rule.aggregate = 'count_unique(tags[sentry:user])';
+        }
+      }
     }
   };
 
   handleFilterUpdate = query => {
     this.setState({query});
   };
 
+  handleAggregateUpdate = (_index: number, column: Column) => {
+    const string_aggregate = generateFieldAsString(column);
+    this.setState({aggregate: string_aggregate});
+    // TODO: This should bind differently? Its used for when the rule saves
+    this.props.rule.aggregate = string_aggregate;
```

Comment:
Mutating props in place is something you want to avoid as react can't keep the rendered state in sync. Instead you would solve this problem using component state, and then read that state when saving.

Reply:
Echoing what mark said - mutating props like this will 1) not cause a re-render like `setState` does, so UI that renders because of `props.rule` will not update and 2) this also potentially cause issues with certain perf optimizations (e.g. in lifecycle methods or using `PureComponent`) where there are strict equality checks with `prop`. In this case if we were to compare (`===`) the previous `rule` prop against current `rule` in a lifecycle method, this would return true because it's comparing the object references vs values.